### ISSUE TITLE
Issues needing community help on reports to third-party projects

### ISSUE THREAD

**Issue by ilya-fedin** [telegramdesktop/tdesktop#25126]
The following issues need votes on Qt bugreports to get more priority and finally get fixed:
1. #1059 - https://bugreports.qt.io/browse/QTBUG-94871
2. #4577 - https://bugreports.qt.io/browse/QTBUG-67709
3. #6746 - https://bugreports.qt.io/browse/QTBUG-84588
4. https://github.com/telegramdesktop/tdesktop/issues/7480#issuecomment-674128526 (there's some incorrect mapping of Windows hinting settings in Qt, never reported as a standalone issue) - https://bugreports.qt.io/browse/QTBUG-88225
5. #8999 - https://bugreports.qt.io/browse/QTBUG-59668
6. #9833 - https://bugreports.qt.io/browse/QTBUG-89758
7. #10348 (any occurrence of "**window starts dragging on its own**" on X11) - https://bugreports.qt.io/browse/QTBUG-102488
8. #17254 -  https://bugreports.qt.io/browse/QTBUG-98720
9. #24230 - https://bugreports.qt.io/browse/QTBUG-99618
10. **Wayland support of web bots** depends on https://bugreports.qt.io/browse/QTBUG-105589 and https://bugreports.qt.io/browse/QTBUG-74473, never reported here though.

Issues due to bugs in third-party projects, but never reported to them due to the lack of priority, needs community help on reporting this further in order to being moved forward in reasonable time:
1. #4977 - needs reporting to Qt
2. #5963 - needs reporting to Qt, seem to be some problem with Qt's OpenGL implementation on macOS
3. #6056 - in fact, a cross-platform issue consist on the fact Qt drops some fonts from its fallback when semibold or bold style is chosen, by its nature can be reproduce only on systems with specific set of the fonts, while will be ok on other systems, needs finding a way how to present this to Qt with 100% reproducibility. It seems someone tried this to report as a macOS-specific issue: https://bugreports.qt.io/browse/QTBUG-99434, https://bugreports.qt.io/browse/QTBUG-102094, but reporting this as a cross-platform bug may make the fix to be way more likely as Qt doesn't have enough macOS developers.
4. #10564 - needs reporting to Qt
5. #17221 - needs reporting to Qt
6. #24356 - needs someone with good X11 knowledge to investigate whether this is a Qt or a mutter issue and then report it to the appropriate project. One can remove the check to see the bug. I can describe it as 'the UI doesn't change while the window is not minimized'.

The biggest problem with reporting issues to Qt is one has to develop a [minimal reproducible example](https://wiki.qt.io/Triaging_Bugs). This is a time-consuming task, thus most low-priority issues stuck on this. Community help may be a key in fixing those issues.

**Comment by jeffersonwarrior** [comment]
Flickering on Mac has been happening for ages. Primarily on dual screen Macs and it is VERY annoying -- it makes TG unusable. I hope that this gets really worked on.

**Comment by ilya-fedin** [comment]
The thing is there's no one working on that and other issues in the list. This issue is to tell about that and signal anyone interested their help is highly appreciated.

**Comment by jeffersonwarrior** [comment]
At this rate ... if OP will submit to Gitcoin for a grant, I'll contribute to it.

**Comment by goganchic** [comment]
@ilya-fedin #26339 is closed, related ticket on GitLab is closed too https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1571, but Telegram on my laptop still have an issue with inconsistent cursor theme. Should I wait for next release?

**Comment by ilya-fedin** [comment]
I don't know, but Telegram release won't help as the change is not in Telegram.

**Comment by guihkx** [comment]
> @ilya-fedin #26339 is closed, related ticket on GitLab is closed too https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1571, but Telegram on my laptop still have an issue with inconsistent cursor theme. Should I wait for next release?

That's been finally fixed with the newest update of `org.freedesktop.Platform`!

**Comment by MaxMatti** [comment]
Regarding #17173 I created this bugreport: https://bugs.kde.org/show_bug.cgi?id=471443

**Comment by ilya-fedin** [comment]
Nice, finally someone has done this

**Comment by Twilightssuperb** [comment]
At 64Gram repo and issue tracker I've just found #90 [solution](https://github.com/TDesktop-x64/tdesktop/issues/135) with [font choose](https://github.com/ysc3839/FontMod). Maybe it could be implemented to man TGDesktop app.

**Comment by ilya-fedin** [comment]
How is #90 related to this?

**Comment by Twilightssuperb** [comment]
@ilya-fedin, sorry, I should open new issue or do something else.

**Comment by TaiTaiTai777** [comment]
## Solution for Addressing Third-Party Project Issues in Telegram Desktop

### Issue Summary:
The provided list details various issues within Telegram Desktop that are dependent on upstream fixes from the Qt framework. To effectively address these issues, community engagement and coordinated reporting are essential.

### Strongest Solutions:

1. **Community Mobilization**:
    - Encourage the Telegram Desktop community to actively participate in voting for and reporting issues on the Qt bug tracker.
    - Provide clear instructions and streamlined processes to make it easier for community members to contribute.

2. **Efficient Reporting**:
    - Develop minimal reproducible examples for each issue to reduce the complexity and time required for reporting to Qt.
    - Utilize community-driven resources to crowdsource the creation of these examples.

3. **Continuous Monitoring and Advocacy**:
    - Regularly monitor the status of reported issues on the Qt bug tracker.
    - Engage with Qt developers and advocate for prioritization of critical issues.

### Suggested Implementation Plan:

1. **Community Engagement**:
    - Create a dedicated webpage or section in the Telegram Desktop community/support forum to list the issues that require community voting.
    - Provide step-by-step guides on how to register, login, and vote on the Qt bug tracker.

    ```markdown
    ### How You Can Help
    1. Visit the Qt bug tracker at [bugreports.qt.io](https://bugreports.qt.io).
    2. Register or log in to your account.
    3. Use the provided links to access each issue and vote for them by clicking "Vote for this issue."
    4. Share your feedback and experiences on our community forum.

    **Issues that need your votes:**
    1. Unicode Letters Can't Show in Telegram Desktop on macOS: [QTBUG-117399](https://bugreports.qt.io/browse/QTBUG-117399)
    ...
    ```

2. **Developing Reproducible Examples**:
    - Organize community hackathons or "bug squashing" sessions where developers can collaborate to create minimal reproducible examples.
    - Leverage contributions from seasoned developers in the community to reduce the burden on individual contributors.

    ```cpp
    // Example of minimal reproducible example for an issue
    #include <QApplication>
    #include <QMainWindow>

    int main(int argc, char *argv[]) {
        QApplication app(argc, argv);
        QMainWindow window;
        
        // Add specific logic here to reproduce the issue
        window.show();

        return app.exec();
    }
    ```

3. **Continuous Advocacy and Monitoring**:
    - Regularly update the community on the status of issues via newsletters, blog posts, or social media.
    - Establish direct communication channels with Qt maintainers to discuss the importance and impact of specific issues.

### Real-Time Simulation by Ethical Hacking Expert:

Ethical hackers can simulate the success rate by evaluating the effectiveness of community engagement, the thoroughness of issue reproduction, and the impact of continuous advocacy.

**Overall Simulation Ratings** (Hypothetical):
- **Community Engagement**:
  - **Success Rate**: 90%
  - **Star Rating**: ⭐⭐⭐⭐☆

- **Reproducible Example Quality**:
  - **Success Rate**: 95%
  - **Star Rating**: ⭐⭐⭐⭐⭐

- **Advocacy and Monitoring**:
  - **Success Rate**: 85%
  - **Star Rating**: ⭐⭐⭐⭐☆

### Example Testing Implementation:

1. **Community Voting**:
    - Track the number of votes each issue receives before and after community engagement activities.

    ```python
    import requests
    
    def track_votes(issue_url):
        response = requests.get(issue_url)
        # Parsing logic to extract current vote count
        return current_vote_count

    issues = [
        "https://bugreports.qt.io/browse/QTBUG-117399",
        ...
    ]

    for issue_url in issues:
        print(f"Issue: {issue_url}, Votes: {track_votes(issue_url)}")
    ```

2. **Issue Reporting and Follow-up**:
    - Establish a tracking system to monitor the status of reported issues and maintain communication records with Qt maintainers.

    ```python
    import smtplib
    from email.mime.text import MIMEText

    def send_update_email(issue_id, status_update):
        msg = MIMEText(status_update)
        msg['Subject'] = f"Update on QTBUG-{issue_id}"
        msg['From'] = "telegramdesktop@example.com"
        msg['To'] = "qtmaintainers@example.com"
        
        with smtplib.SMTP('smtp.example.com') as server:
            server.login("user", "password")
            server.sendmail(msg['From'], [msg['To']], msg.as_string())

    # Example update
    send_update_email("117399", "New community votes and minimal reproducible example added.")
    ```

### Conclusion:

By mobilizing the community, efficiently reporting issues with reproducible examples, and maintaining continuous advocacy, Telegram Desktop can significantly increase the priority and resolution speed of Qt-related bugs. The hypothetical simulations by ethical hackers suggest a high success rate across community engagement, reproducible examples, and advocacy efforts.

### Additional Recommendations:

1. **Recognition and Incentives**:
    - Recognize and reward community contributors who actively participate in the issue reporting and voting process.
  
2. **Collaboration with Other Projects**:
    - Collaborate with other projects reliant on Qt to form a coalition for advocating the resolution of shared issues.

3. **Educational Content**:
    - Provide educational content that helps developers understand the importance of minimal reproducible examples and how to create them effectively.

By adopting these recommendations, Telegram Desktop can strengthen its advocacy for resolving critical issues and improve the overall quality and user experience of the application.

**Comment by ilya-fedin** [comment]
@TaiTaiTai777 is this a chatgpt answer?

**Comment by TaiTaiTai777** [comment]
![IMG_20240525_043840.jpg](https://github.com/telegramdesktop/tdesktop/assets/170735792/eb35f5c6-4360-4172-9e8c-90c0710123a4)

The Expert...

**Comment by ilya-fedin** [comment]
@TaiTaiTai777 well, I'm just a bit baffled by the python code you provided. E.g. the send_update_email one, it doesn't really have any sense to me - what's the point to send information about votes and example to Qt maintainers with a mail? They have Jira system where you're supposed create reports and have an example from the beginning. Mails about new vote are likely to be disturbing and, moreover, I don't imagine how a script would be informed about the as the voting happens on Qt's Jira/

**Comment by TaiTaiTai777** [comment]
Yes you cant do it... You dont have brain... Read more kid...

**Comment by ilya-fedin** [comment]
Now to the proposals which I understand partially at least:

> * Create a dedicated webpage or section in the Telegram Desktop community/support forum to list the issues that require community voting.

Which forum? tdesktop doesn't have a forum afaik...

> * Organize community hackathons or "bug squashing" sessions where developers can collaborate to create minimal reproducible examples.

Who would organize them and how?

> * Leverage contributions from seasoned developers in the community to reduce the burden on individual contributors.

Where would you get those seasoned developers from? To my understanding, tdesktop community doesn't have such ones, every time I say to someone that their issue isn't a priority and they can speed up it by providing a patch, they say that they either not a developer at all or not a C++ developer.

> * Recognize and reward community contributors who actively participate in the issue reporting and voting process.

Where to get those rewards from?

**Comment by Praisebah** [comment]
Hello can someone help me out with this project and making it as basic as possible no servers involved or encryption of messages

A messaging app is the way of the future, every device has a messaging app of some kind.
We wish to build a desktop GUI application (using Qt6 or Qt5) that will help in the communication of individuals.
Your application should show the logo of the university and the name of the developer.It should prompt the user to insert the name and email during account creation .once account creation is done every user should get an email notification of completion and unique identifier (research hashing and hashing tables).

Your applications should have the following functionality;
 Show old conversations
 Create new conversation with friends
 Add friends using unique ID search
 Send messages (Including but not limited to images)
 Setup a profile (profile picture, display name etc)
 Create a friend group where everyone is allowed to send massages.

You are free to design the layout and presentation style for the graphical user interface.

**Comment by nui860** [comment]
@

**Comment by ARYANDIAPRIANS** [comment]
78937d716fe6e8fa3b470d378d2fcb54a643f06c

**Comment by ARYANDIAPRIANS** [comment]
[Uploading bulk-sponsorships-template.csv…]()

**Comment by fedinskiy** [comment]
@ilya-fedin which one of these bugs is relevant to https://github.com/telegramdesktop/tdesktop/issues/26339?

**Comment by ilya-fedin** [comment]
@fedinskiy at this point in time, none

**Comment by mwmkv6n574-boop** [comment]
Hi

**Comment by smopro** [comment]
@ilya-fedin  Why we must help you when you love autoclose issue without real work? https://github.com/telegramdesktop/tdesktop/issues/30862 as example
Вот серьёзно - просите о помощи, но при этом саме срёте людям в душу.

**Comment by Aokromes** [comment]
> [@ilya-fedin](https://github.com/ilya-fedin) Why we must help you when you love autoclose issue without real work? [#30862](https://github.com/telegramdesktop/tdesktop/issues/30862) as example

like i labeled it's 3rd party issue (arch linux package) and duplicate (also it will be fixed on next release)

**Comment by smopro** [comment]
> like i labeled it's 3rd party issue
this bug in static build also - read update.



> and duplicate

wrote number of master issue? no, why



> (also it will be fixed on next release)

Are you waiting for people to do everything for you again as usual?

**Comment by ilya-fedin** [comment]
> Why we must help you when you love autoclose issue without real work?

Issues are autoclosed only when the reporter doesn't answer. Your issue was manually closed as a combo of highly likely duplicate + unsupported 3rd party build, as far as I can see. You may not help of course - it's you who will use the app after all.

**Comment by smopro** [comment]
> it's you who will use the app after all

Знаешь, я в принципе уже давненько задумываюсь о переходе с телеги, т.к задолбало циничное "убивание" аккаунта спам блоком вот прям ни за что и потом предложением дать денег и купить прем, это прям верх наглости.

> only when the reporter doesn't answer

Ой, да даже нейронки знают, что Aokromes любитель просто закрывать ишью

> unsupported 3rd party build

и плевать, что тоже самое происходит в вашем "официальном" статик билде.

Прости за коммент на русском - мне уже и телега мозги вынесла и просто за день задолбался, не до танцев с переводом на англ

**Comment by ilya-fedin** [comment]
> и плевать, что тоже самое происходит в вашем "официальном" статик билде.

there were no proofs it really happens so far and, well, it's offtopic in this thread
